cabal-install/main/Main.hs

```diff
@@ -288,6 +288,30 @@ mainWorker args = do
                                   ++ "\ncompiled using version "
                                   ++ display cabalVersion
                                   ++ " of the Cabal library "
+                                  ++ flags_enabled
+      where
+        flags_enabled | null enabledFlags   = "\nall flags disabled"
+                      | otherwise = "\nflags enabled: " ++ intercalate ", " enabledFlags
+        enabledFlags =
+#ifdef CABAL_FLAG_NATIVEDNS
+            "native-dns" :
```

Comment:
Maybe even render them as `-fnative-dns`, `-flib`, etc.

Reply:
@23Skidoo then you could also enumerate all known flags explicitly everytime using the more explicit `--flags=` syntax:

`--flags="-native-dns +lib -monolithic"`

Reply:
That's even better!